src/test/unit/utils/RandomTest.cpp

```diff
@@ -31,246 +31,82 @@
-    611254689,  1326824505, 926723433,  1895605687, 1448376866, 212908541,
-    941010526,  1047113264, 1584402020, 1659427688, 2127915429, 471804235,
-    83700688,   883702914,  1702189562, 1931715164, 672974791,  2043878592,
-    1311021947, 637136544,  1990201214, 2128228362, 946861166,  2091436239,
-    216042476,  2041101890, 1728907825, 153287276,  1886925555, 2138321635,
-    273154489,  350696597,  1317662492, 1199877922, 98818636,   618555710,
-    1412786463, 1039829162, 1665668975, 849704836,  551773203,  1646100756,
-    1321509071, 635473891,  382320022,  876214985,  419705407,  1055294813,
-    772609929,  1730727354, 1692431357, 615327495,  1711472069, 491808875,
-    559280086,  1927514545, 385427118,  140704264,  2080801821, 124869025,
-    131542251,  206472663,  475565622,  1449204744, 1406350585, 574384258,
-    2067760454, 671653401,  1614213421, 1585945781, 1521358237, 18502976,
-    1084562889, 695383660,  653976867,  1466882911, 1571598645, 1073682275,
-    374694077,  196724927,  656925981,  2067125434, 812052422,  220914402,
-    411450662,  1371332509, 945300,     796877780,  1512036773, 2081747121,
-    921746805,  1643579024, 140736136,  1397312428, 945300120,  1547086722,
-    1971696686, 865576927,  71256475,   1438426459, 304039060,  1592614712,
-    1456929435, 1388601950, 140514724,  2110906303, 708001213,  1712113369,
-    1037104930, 1082695290, 1908838296, 1694030911, 1002337077, 573407071,
-    1914945314, 1413787739, 1944739580, 1915890614, 63181871,   1309292705,
-    1850154087, 984928676,  805388081,  1990890224, 234757456,  1750688202,
-    1390493298, 58970495,   468781481,  1461749773, 1497396954, 772820541,
-    906880837,  806842742,  13938843,   1047395561, 770265397,  721940057,
-    612025282,  1807370327, 1804635347, 373379931,  1353917590, 659488776,
-    946787002,  1121379256, 2073276515, 744042934,  889786222,  2136458386,
-    2053335639, 592456662,  973903415,  711240072};
+using namespace std;
 
 TEST(RandomTest, Seeding) {
-  // make sure the global instance is seeded from time()
-  // in the test situation, we can be sure we were seeded less than 100000
-  // seconds ago make sure random number system is initialized by creating a
-  // random object. Use the object to make sure the compiler doesn't complain
-  // about an unused variable
+  {
   Random r;
-  UInt64 x = r.getUInt64();
+  auto x = r.getUInt32();
   ASSERT_TRUE(x != 0);
+  }
 
   // test getSeed
   {
     Random r(98765);
     ASSERT_EQ(98765U, r.getSeed());
   }
+
+  { // seed & equals
+  Random r1(1); 
+  Random r2(2);
+  ASSERT_NE(r1.getUInt32(), r2.getUInt32());
+  ASSERT_NE(r1, r2) << "Randoms with different seed must be different";
+
+  Random r1b(1);
+  ASSERT_NE(r1, r1b) << "different steps";
+  auto x = r1b.getUInt32();
+  ASSERT_EQ(r1, r1b) << "Randoms with same seed must be the same." << x; 
+  }
+
+  { //MAX_INT seed
+  Random r(-1);
+  ASSERT_EQ(r(), 111299194u);
```

Comment:
@ctrl-z-9000-times 

I could not reproduce your test/fail: 

> If you seed the custom random number generator with -1 then it always returns zero. What are the odds?
Random rng(-1);
while( rng() == 0 ) {}
// This loops forever ...